pkg/node/node_controller.go

```diff
@@ -77,20 +73,47 @@ func (b *DirectCSIDriveListener) Update(ctx context.Context, old, new *v1alpha1.
 			return nil
 		}
 
+		if new.DriveStatus == v1alpha1.Unformatted && new.RequestedFormat.Filesystem == "" {
+			return status.Errorf(codes.InvalidArgument, "Filesystem has to be specified for formatting")
+		}
+
+		isForceOptionSet := new.RequestedFormat.Force
+		fsType := new.RequestedFormat.Filesystem
+		copiedDrive := new.DeepCopy()
+		if fsType != "" {
+			if new.Filesystem != "" && !isForceOptionSet {
+				return status.Errorf(codes.InvalidArgument, "Need to set `force` option to override the format of this drive: %s", old.ObjectMeta.Name)
+			}
+			if new.Mountpoint != "" {
+				// Check and unmount if the drive is already mounted
+				if err := UnmountIfMounted(new.Mountpoint); err != nil {
```

Comment:
we should first check if there are any volumes being used from this drive before unmounting.